### PR TITLE
bench: add fibonacci/factorial comparison with revm

### LINKED ISSUE #215: Clean up benchmarks and add them under `bench/`
The benchmarks are in the [`bench` branch](https://github.com/lambdaclass/evm_mlir/tree/bench).

### PR DESCRIPTION
**Pull request by MegaRedHand**
Closes #215 .

This PR adds hyperfine benchmarks.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -49,3 +49,27 @@ fmt:
...
+	@printf "%s" "revm_factorial result: "
+	@target/release/revm_factorial 1
+	hyperfine -w 5 -r 10 -N \
+		-n "evm_mlir_factorial" "target/release/evm_mlir_factorial 100000" \
+		-n "revm_factorial" "target/release/revm_factorial 100000"
+
+	@printf "%s" "evm_mlir_fibonacci result: "
+	@target/release/evm_mlir_fibonacci 1
+	@printf "%s" "revm_fibonacci result: "
+	@target/release/revm_fibonacci 1
```
Comment (MegaRedHand) [inline Makefile:72]: Can you add an empty `echo` before and after these results?
```suggestion
	@echo
	@printf "%s" "evm_mlir_fibonacci result: "
	@target/release/evm_mlir_fibonacci 1
	@printf "%s" "revm_fibonacci result: "
	@target/release/revm_fibonacci 1
	@echo
```

Same for the ones above

File: bench/revm_comparison/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+[package]
+name = "revm_factorial"
```
Comment (MegaRedHand) [inline bench/revm_comparison/Cargo.toml:2]:
```suggestion
name = "revm_comparison"
```

File: bench/revm_comparison/src/evm_mlir_factorial.rs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+use revm_comparison::run_with_evm_mlir;
+use std::env;
+
+fn main() {
+    const PROGRAM: &str = "7f0000000000000000000000000000000000000000000000000000000000000080600260025b8215603b57906001018091029160019003916025565b9150505f5260205ff3";
+    let args: Vec<String> = env::args().collect();
+    let runs = &args[1];
```
Comment (MegaRedHand) [inline bench/revm_comparison/src/evm_mlir_factorial.rs:7]: We can use `nth(1)` here:
```suggestion
    let runs = env::args().nth(1).unwrap();
```

**Review by MegaRedHand — COMMENTED** [review]
LGTM

**Review by Oppen — APPROVED** [review]
LGTM. It may be a good idea to `black_box` the arguments and result of runs, but given it's a lot of logic it's unlikely to get inlined anyway.

**Pull request merged**